k2/csrc/properties.cc

```diff
@@ -78,6 +79,41 @@ bool HasSelfLoops(const Fsa &fsa) {
   return false;
 }
 
+bool CheckCycles(StateId s, std::vector<bool> visited, std::vector<bool> back_arc,
+                 std::unordered_set<StateId> *adj) {
+  if(visited[s] == false) {
+    visited[s] = true;
+    back_arc[s] = true;
+
+    for (auto i = adj[s].begin(); i != adj[s].end(); ++i) {
+      if (!visited[*i] && CheckCycles(*i, visited, back_arc, adj) )
+        return true;
+      else if (back_arc[*i])
+        return true;
+    }
+  }
+  back_arc[s] = false;
+  return false;
+}
+
+// Detect cycles using DFS traversal
+bool IsAcyclic(const Fsa &fsa) {
+  StateId num_states = fsa.NumStates();
+  std::vector<bool> visited(num_states, false);
+  std::vector<bool> back_arc(num_states, false);
+
+  std::unordered_set<StateId> *adj = new std::unordered_set<StateId>[num_states];
```

Comment:
`adj` is not freed and memory is leaked here.

Reply:
I defined 'adj' as non-pointer and removed the 'new' operation.